config/federatedtypes/serviceaccount.yaml

```diff
@@ -0,0 +1,17 @@
+apiVersion: core.federation.k8s.io/v1alpha1
+kind: FederatedTypeConfig
+metadata:
+  name: serviceaccounts
+spec:
+  target:
+    version: v1
+    kind: ServiceAccount
+  namespaced: true
+  comparisonField: ResourceVersion
```

Comment:
Why here using `ResourceVersion` as the comparisonField? I checked the difference `ResourceVersion` and `Generation` at https://stackoverflow.com/questions/47100389/what-is-the-difference-between-a-resourceversion-and-a-generation/47101418 , but still not clear what is the best practise for this value.

FYI @marun

Reply:
@danehans I saw that you have updated this value to `Generation` in follow up commits, can you please show some detail for the guidance of when to use `Generation` and when to use `ResourceVersion`? Thanks.